packages/utils/telemetry-utils/src/telemetryTypes.ts

```diff
@@ -0,0 +1,137 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ITelemetryBaseLogger, TelemetryEventCategory } from "@fluidframework/common-definitions";
+
+/**
+ * Property types that can be logged.
+ *
+ * @remarks Logging entire objects is considered extremely dangerous from a telemetry point of view because people can
+ * easily add fields to objects that shouldn't be logged and not realize it's going to be logged.
+ * General best practice is to explicitly log the fields you care about from objects.
+ */
+export type TelemetryEventPropertyTypeExt =
+    | string
+    | number
+    | boolean
+    | undefined
+    | (string | number | boolean)[];
```

Comment:
@vladsud - were you suggesting we support `null` in here too? Kind of hard to imagine needing it (a list of optional numbers or something...?), but in the interest of being a flexible API seems worth including. I know we shouldn't do `undefined` since `JSON.stringify([undefined]) === "[null]"`.